src/compiler/checker.ts

```diff
@@ -365,6 +366,13 @@ namespace ts {
             getDeclaredTypeOfSymbol,
             getPropertiesOfType,
             getPropertyOfType: (type, name) => getPropertyOfType(type, escapeLeadingUnderscores(name)),
+            getPrivateIdentifierPropertyOfType: (leftType: Type, right: PrivateIdentifier) => {
+                if (!isParseTreeNode(right)) {

```

Comment:
This won't work with custom transforms where someone may set an `original` pointer on the node. A better approach would be:
```ts
right = getParseTreeNode(right, isPrivateIdentifier) || Debug.fail("Cannot get properties using a private identifier that cannot be resolved to a parse-tree node.");
```

Reply:
Or since this function can return `undefined`, just return `undefined` if you couldn't find a parse tree node (this is the behavior of other methods similar on TypeChecker).

Reply:
Thanks, I fixed to use `getParseTreeNode` and to return `undefined` instead of asserting.